sdk/python/feast/feature_store.py

```diff
@@ -1017,6 +1013,30 @@ def _augment_response_with_on_demand_transforms(
                     ] = GetOnlineFeaturesResponse.FieldStatus.PRESENT
         return OnlineResponse(GetOnlineFeaturesResponse(field_values=result_rows))
 
+    def _get_feature_views_to_use(
+        self,
+        features: Optional[Union[List[str], FeatureService]],
+        allow_cache=False,
+        hide_dummy_entity: bool = True,
+    ) -> List[FeatureView]:
+
+        passed_in_feature_views = (
+            {view.name: view for view in features.feature_views}
+            if isinstance(features, FeatureService)
+            else {}
+        )
+
+        all_feature_views = [
```

Comment:
It's not immediately obvious to me what is happening here. We are querying the registry using `_list_feature_views` then filtering based on `passed_in_feature_views`, so why do we need `passed_in_feature_views` then?

Reply:
the logic here has changed in the new commits.